lib/tools/system-calls.js

```diff
@@ -855,26 +861,41 @@ systemCallMethods.reboot = async function reboot (retries = DEFAULT_ADB_REBOOT_R
   });
 };
 
+/**
+ * Restart adb but only if no devices are connected
+ */
+systemCallMethods.restartAdbIfDevicesOffline = async function restartAdbIfDevicesOffline () {
+  const connectedDevices = await this.getConnectedDevices();
+  if (connectedDevices.length === 0) {
```

Comment:
this will not work for parallel tests

Reply:
What do you recommend?

Reply:
we should rely on the actual command output instead